src/batch.rs

```diff
@@ -0,0 +1,61 @@
+//! # Batch
+//!
+//! This module contains types for batch operations. A batch operation is a
+//! collection of write operations (puts and/or deletes) that are applied
+//! atomically to the database.
+
+/// A batch of write operations (puts and/or deletes). All operations in the
+/// batch are applied atomically to the database. If multiple operations appear
+/// for a a single key, the last operation will be applied. The others will be
+/// dropped.
+///
+/// # Examples
+/// ```rust,no_run,compile_fail
+/// use slatedb::batch::{WriteBatch, WriteOp};
+///
+/// let object_store = Arc::new(LocalFileSystem::new());
+/// let db = Db::open("path/to/db".into(), object_store).await;
+///
+/// let mut batch = WriteBatch::new();
+/// batch.put(b"key1", b"value1");
+/// batch.put(b"key2", b"value2");
+/// batch.delete(b"key3");
+///
+/// db.write(batch).await;
+/// ```
+///
+/// Note that the `WriteBatch` has an unlimited size. This means that batch
+/// writes can exceed `l0_sst_size_bytes` (when `WAL` is disabled). It also
+/// means that WAL SSTs could get large if there's a large batch write.
+pub struct WriteBatch {
+    pub(crate) ops: Vec<WriteOp>,
+}
+
+impl Default for WriteBatch {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+pub enum WriteOp {
+    Put(Vec<u8>, Vec<u8>),
```

Comment:
Do y'all think we should use a skiplist like we do in KVTable? It would make iterator support easier (required for txns). It would lose the full history of put/delete/put operations on a single key. It comes with some extra memory overhead.

I'm inclined to use skiplist. WDYT?

Reply:
Txn can maintain its own local buffer. Taking SSI as an example, when our validation passes, we call `write_batch` to write to LSM.

Reply:
So are you suggesting to keep it as a Vec?

Reply:
No big preference. What about `Byte`？ it just popped into my head, since we still need `Bytes::copy_from` when inserting into the Memtable, why not do it initially?

Reply:
But I think at least `WriteBatch` does not need to maintain data structures like skiplist.

Reply:
I took a shot at `Bytes`: d36b55e224997169d5b37b03ceda778b43f5fdbb

In doing so, I also updated the `mem_table` code to use `Bytes` instead of `&[u8]`. So, now keys and values get converted from slice to Bytes on get/put/delete, and the Bytes object is cloned/passed everywhere else down the call stack.